website/docs/reference/global-configs/warnings.md

```diff
@@ -2,129 +2,216 @@
+  - [dbt-core's types.py file](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py), where each class name that inherits from `WarnLevel` corresponds to a warning name (e.g. `AdapterDeprecationWarning`, `NoNodesForSelectionCriteria`).
+  - Downloading the JSON output logs from a run and searching for the warning.
+- The `error` parameter can be set to `"all"` or `"*"` to treat all warnings as exceptions (errors), or to a list of specific warning names to treat as exceptions. This behavior is the same as using the `--warn-error` flag.
+  
+  When `error` is set to `"all"` or `"*"`, the optional `warn` parameter can be set to exclude specific warnings from being treated as exceptions &mdash; similar to the [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) flag.
+- Use the `silence` parameter to ignore warnings. For example, to silence deprecation warnings or certain warnings you want to ignore across your project, you can specify them in the `silence` parameter. This is useful in large projects where certain warnings aren't critical and can be ignored to keep the noise low and logs clean.
 
-- The `error` parameter can be set to `"all"` or `"*"` to treat all warnings as exceptions, or to a list of specific warning names to treat as exceptions. When `error` is set to `"all"` or `"*"`, the optional `warn` parameter can be set to exclude specific warnings from being treated as exceptions.
+</VersionBlock>
 
-- Use the `silence` parameter to ignore warnings through project flags, without needing to re-specify the silence list every time. For example, to silence deprecation warnings or certain warnings you want to ignore across your project, you can specify them in the `silence` parameter. This is useful in large projects where certain warnings aren't critical and can be ignored to keep the noise low and logs clean.
+## Configuration
 
+You can configure warnings as errors or which warnings to silence, by setting the config through command flags, environment variables, or configuration files like `dbt_project.yml` or `profiles.yml`.
 
-<File name='dbt_project.yml'>
+<VersionBlock lastVersion="1.7"> 
 
-```yaml
-name: "my_dbt_project"
-tests:
-  +enabled: True
-flags:
-  warn_error_options:
-    error: # Previously called "include"
-    warn: # Previously called "exclude"
-    silence: # To silence or ignore warnings
-      - NoNodesForSelectionCriteria
-```
+- Promote all warnings to errors using `{"include": "all"}` or `--warn-error` flag.
+- Promote some warnings to errors using `include`.
+- Exclude warnings from being treated as errors using `exclude`.
 
-</File>
+In the following example, we're ignoring the [`NoNodesForSelectionCriteria` warning](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.py#L1227) in the `profiles.yml` file by adding it to the `exclude` parameter:
 
-```text
-dbt --warn-error-options '{"error": "all"}' run
-...
-```
+  <File name='profiles.yml'>
```

Comment:
let's not show the example in profiles.yml, let's use dbt_project.yml instead

Reply:
@graciegoheen i was going to update this to project.yml but i read in the [1.8 upgrade guide](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.8#quick-hits) that "Custom defaults of [global config flags](https://docs.getdbt.com/reference/global-configs/about-global-configs) should be set in the flags dictionary in [dbt_project.yml](https://docs.getdbt.com/reference/dbt_project.yml), instead of in [profiles.yml](https://docs.getdbt.com/docs/core/connect-data-platform/profiles.yml). Support for profiles.yml has been deprecated." so i interpreted that to mean that any examples for 1.7 and lower should show profiles.yml -- is that right?

Reply:
ooooooh yep i just missed this versioning block :/